src/move_basic.cpp

```diff
@@ -457,81 +452,40 @@ void MoveBasic::executeAction(const move_base_msgs::MoveBaseGoalConstPtr& msg)
-            double requestedYaw = atan2(linear.y(), linear.x());
-            if (reverseWithoutTurning) {
-                if (requestedYaw > 0.0) {
-                    requestedYaw = -M_PI + requestedYaw;
-                }
-                else {
-                    requestedYaw = M_PI - requestedYaw;
-                }
+    if (dist > linearTolerance) {
+        double requestedYaw = atan2(linear.y(), linear.x());
+        if (reverseWithoutTurning) {
+            if (requestedYaw > 0.0) {
+                requestedYaw = -M_PI + requestedYaw;
             }
-
-            if (std::abs(requestedYaw) < angularTolerance) {
-                break;
+            else {
+                requestedYaw = M_PI - requestedYaw;
             }
+        }
+
+        // Initial rotation
+        if (std::abs(requestedYaw) > angularTolerance) {
             if (!rotate(requestedYaw, drivingFrame)) {
                 return;
             }
-            sleep(localizationLatency);
         }
-    }
-
-    // Do linear portion of goal
-    ROS_INFO("MoveBasic: Requested distance %f with linear tolerance of %f", dist,linearTolerance);
-
-    if (std::abs(dist) > linearTolerance) {
-        if (reverseWithoutTurning) {
-            dist = - dist;
-        }
-        if (!moveLinear(goalInDriving, planningFrame, drivingFrame)) {
+        // Linear portion
+        if (!moveLinear(goalInDriving, drivingFrame)) {
             return;
         }
-        {
-            tf2::Transform poseFrameIdFinal;
-            if (!getTransform(baseFrame, frameId, poseFrameIdFinal)) {
-                 ROS_WARN("MoveBasic: Cannot determine robot pose in goal frame");
-            }
-            tf2::Vector3 distTravelled = poseFrameIdFinal.getOrigin() -
-                                         poseFrameId.getOrigin();
-            ROS_DEBUG("MoveBasic: Travelled %f %f\n", distTravelled.x(), distTravelled.y());
-        }
-
-        sleep(localizationLatency);
-    }
 
-    // Final rotation as specified in goal
-    tf2::Transform finalPose;
-    if (!getTransform(baseFrame, drivingFrame, finalPose)) {
-         abortGoal("MoveBasic: Cannot determine robot pose for final rotation");
-         return;
-    }
-
-    getPose(finalPose, x, y, yaw);
-    rotate(goalYaw - yaw, drivingFrame);
-
-/*
-    sleep(10);
-    // Final sanity check
-    {
-        tf2::Transform poseFrameIdFinal;
-        if (!getTransform(baseFrame, frameId, poseFrameIdFinal)) {
-             ROS_WARN("Cannot determine robot pose in goal frame");
+        // Final rotation
+        if (std::abs(goalYaw - (yaw + requestedYaw)) > angularTolerance) {
+            if (!rotate(goalYaw - (yaw + requestedYaw), drivingFrame)) {
+                return;
+            }
         }
-        tf2::Vector3 distTravelled = poseFrameIdFinal.getOrigin() -
-                                     poseFrameId.getOrigin();
-        ROS_DEBUG("Travelled %f %f\n", distTravelled.x(), distTravelled.y());
```

Comment:
Wouldn't this still be useful for debug?

Reply:
Good point. I will add the debug stuff